src/lvl_script.c

```diff
@@ -4119,6 +4117,90 @@ TbBool script_level_up_creature(PlayerNumber plyr_idx, long crmodel, long criter
     return true;
 }
 
+/**
+ * Cast a spell on a creature which meets given criteria.
+ * @param plyr_idx The player whose creature will be affected.
+ * @param crmodel Model of the creature to find.
+ * @param criteria Criteria, from CreatureSelectCriteria enumeration.
+ * @param magic_caster_splvl encoded bytes: magic id, caster player index and spell lvl (MSD order)
+ * @return TbResult whether the spell was successfully cast
+ */
+TbResult script_cast_spell_on_creature(PlayerNumber plyr_idx, long crmodel, long criteria, long magic_caster_splvl)
+{
+    struct Thing *thing = script_get_creature_by_criteria(plyr_idx, crmodel, criteria);
+    if (thing_is_invalid(thing)) {
+        SYNCDBG(5,"No matching player %d creature of model %d found to cast spell on.",(int)plyr_idx,(int)crmodel);
+        return false;
+    }
+    
+    // get spell type from left octet
+    PowerKind pwkind = magic_caster_splvl >> 16;
+    // get caster PlayerNumber from middle octet
+    PlayerNumber caster =  (magic_caster_splvl >> 8) & ((1 << 8) - 1);
+    // get spell level from right octet
+    long splevel = (magic_caster_splvl & ((1 << 8) - 1));
+    MapSubtlCoord stl_x = thing->mappos.x.stl.num;
+    MapSubtlCoord stl_y = thing->mappos.y.stl.num;
+    switch (pwkind)
+    {
+      case PwrK_HEALCRTR:
+        return magic_use_power_heal(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_SPEEDCRTR:
+        return magic_use_power_speed(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_PROTECT:
+        return magic_use_power_armour(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_CONCEAL:
+        return magic_use_power_conceal(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_DISEASE:
+        return magic_use_power_disease(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_CHICKEN:
+        return magic_use_power_chicken(caster, thing, 0, 0, splevel, PwMod_CastForFree);
+      case PwrK_SLAP:
+        return magic_use_power_slap_thing(caster, thing, PwMod_CastForFree);
+      case PwrK_CALL2ARMS:
+        return magic_use_power_call_to_arms(caster, stl_x, stl_y, splevel, PwMod_CastForFree);
+      case PwrK_LIGHTNING:
+        return magic_use_power_lightning(caster, stl_x, stl_y, splevel, PwMod_CastForFree);
+      case PwrK_CAVEIN:
+        return magic_use_power_cave_in(caster, stl_x, stl_y, splevel, PwMod_CastForFree);
+      case PwrK_MKDIGGER:
```

Comment:
Funny, these spells aren't meant to be cast at units of course, but it works, so why not I guess. Needs proper documentation.

Reply:
Yeah magic_use_power_on_thing does not list them. But I thought it would be great if player could do this via script.